src/os/linux/local/mode/ntp.pm

```diff
@@ -209,39 +232,166 @@ sub get_ntp_modes {
+                                                                no_quit => 1, );
+
+    my %values = ( ( map { /^\s*(.+?): (.+)$/ } split /\n/, $stdout_timesync ),
+                   ( map { /^(.+?)=(.+)$/ } split /\n/, $stdout_show ),
+                   ( map { /^\s*(.+?): (.+)$/ } split /\n/, $stdout_status ) );
+
+    return "timedatectl not available" unless $values{'NTP service'};
+
+    $values{$_}//='' foreach ('SystemNTPServers', 'FallbackNTPServers', 'ServerAddress', 'ServerName', 'Offset', 'Stratum', 'Packet count', 'System clock synchronized');
+
+    my $active_is_fallback = 0;
+
+    # Primary and fallback servers are initialized, the active server is excluded as it will be initialized later with additional information
+    # A server has either the type 'primary' or 'fallback'
+    # A 'primary' server can have the states 'available', 'synchronized', 'syncing' or 'inactive'
+    # A 'fallback' server can have the states 'unused', synchronized', 'syncing' or 'inactive'
+    foreach (split /\s/, $values{SystemNTPServers}) {
+        next if $values{ServerAddress} && ($_ eq $values{ServerAddress} || $_ eq $values{ServerName});
+
+        $self->{peers}->{$_} = { display => $_, rawstate => 'available', stratum => '', rawtype => 'primary', reach => '', offset => '' };
+        $self->{global}->{peers}++
+    }
+    foreach (split /\s/, $values{FallbackNTPServers}) {
+        if ($values{ServerAddress} && ($_ eq $values{ServerAddress} || $_ eq $values{ServerName})) {
+            $active_is_fallback = 1;
+            next
+        }
+        $self->{peers}->{$_} = { display => $_, rawstate => 'unused', stratum => '', rawtype => 'fallback', reach => '', offset => '' };
+        $self->{global}->{peers}++
+    }
+
+    if ($values{ServerAddress} ne '') {
+        # If there is an active server it is initialized here with all information
+        $values{ServerName} = $values{ServerAddress} if $values{ServerName} eq '';
+        $values{'Offset'} = $1 * $unit_map_timedatectl{$2} if $values{'Offset'} =~ /^(.*?)([a-z]+)$/ && $unit_map_timedatectl{$2};
+
+        $self->{peers}->{$values{ServerAddress}} = { display => $values{ServerName},
+                                                     address => $values{ServerAddress},
+                                                     rawstate => $values{'System clock synchronized'} eq 'yes' ? 'synchronized' : 'syncing',
+                                                     rawtype => $active_is_fallback ? 'fallback' : 'primary',
+                                                     stratum => $values{Stratum},
+                                                     reach => $values{'Packet count'},
+                                                     offset => $values{'Offset'} };
+        $self->{global}->{peers}++;
+    }
+
+    foreach (keys %{$self->{peers}}) {
+        $self->{peers}->{$_}->{rawstate} = 'inactive' if $values{'NTP service'} eq 'inactive';
```

Comment:
At this point I don't see how a primary server could ever still be in `available` rawstate.
Am I right?